### PR TITLE
Add Webmin Suite

### PR DESCRIPTION
**Pull request by akubak**
### Description:
Add Webmin installer script & docs page

### Checklist:
  - [x] The code change is tested and works locally.
  - [x] Script has validation check of the job.
  
#### If pertinent:
  - [x] Created/Updated documentation at `/docs`

### CODE REVIEW

**Comment by Landrash** [comment]
Let's have this merged and if we later feel the need to have less we can remove this in the future.

@ludeeus Can you assist in reviewing this one since you have a better understanding of the current tooling?

File: package/opt/hassbian/suites/webmin.sh [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    rm /var/lib/dpkg/info/apt-show*
+    apt-get -f install apt-show-versions -y
+    apt-get install webmin -y
+    if apt-get install webmin -y; then
+      echo "Fix successfully applied"
+    fi
+  fi
+
+  # Get local IP address for this machine
+  LOCAL_IP=$(hostname -I)
```
Comment (ludeeus) [inline package/opt/hassbian/suites/webmin.sh]: This "LOCAL_IP=$(hostname -I)" returns "xxx.xxx.xxx.xxx "
Can you remove the space or use this instead?
ip_address=$(ifconfig | grep "inet.*broadcast" | grep -v 0.0.0.0 | awk '{print $2}')

File: package/opt/hassbian/suites/webmin.sh [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  # End
+  echo
+  echo "Installation done."
```
Comment (ludeeus) [inline package/opt/hassbian/suites/webmin.sh]: Can you add an validation to this script?
See #121 for inspiration ;)

File: package/opt/hassbian/suites/webmin.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    webmin-show-repo-message
+  else
+    echo "Webmin repository not found, adding it now"
+    sed -i '$a deb https://download.webmin.com/download/repository sarge contrib' /etc/apt/sources.list
+  fi
+  unset SOURCES_LIST
+  unset INSTALL_MODE
+}
+
+function webmin-check-keys {
```
Comment (ludeeus) [inline package/opt/hassbian/suites/webmin.sh]: This function is not in use, this is needed, the script will fail without it.

File: package/opt/hassbian/suites/webmin.sh [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  webmin-show-short-info
+  webmin-show-copyright-info
+  webmin-check-repo
+
+  echo "Updating apt-get"
+  apt-get update
+
+  echo "Checking for Webmin updates"
+  apt-get upgrade webmin -y
+  return 0
```
Comment (ludeeus) [inline package/opt/hassbian/suites/webmin.sh]: The upgrade function should also have a validation step.

File: docs/webmin.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+```
+
+## Upgrade
+```
+$ sudo hassbian-config upgrade webmin
+```
+
+## Additional info
+Default port: 10000
+
```
Comment (ludeeus) [inline docs/webmin.md]: Need more info.
- Where is logging done?
- Are there an config file/dir?
- Are there any service commands that could be useful?

**Review by ludeeus — DISMISSED** [review]
There are some parts that needs to be corrected before I can do a full review of this.
One of the functions are not referenced.
When it installed, the web site was not available on port 10000, with this in the service status:
`Feb 27 20:00:19 hassbian perl[3825]: pam_unix(webmin:auth): authentication failure; logname= uid=0 euid=0 tty= ruser= rhost=  user=root`

File: docs/webmin.md (hunk 5 above)
Comment (Antoni-K) [reply docs/webmin.md]: Will add them when I get home.

Comment (akubak) [reply docs/webmin.md]: Will add this soon

File: package/opt/hassbian/suites/webmin.sh (hunk 4 above)
Comment (akubak) [reply package/opt/hassbian/suites/webmin.sh]: I'll implement these as soon as possible

File: package/opt/hassbian/suites/webmin.sh (hunk 3 above)
Comment (akubak) [reply package/opt/hassbian/suites/webmin.sh]: Thanks, fixed it

File: package/opt/hassbian/suites/webmin.sh (hunk 2 above)
Comment (akubak) [reply package/opt/hassbian/suites/webmin.sh]: OK

**Comment by ludeeus** [comment]
Output when running:
https://hastebin.com/unonulexej.sql
Webmin did not get installed.

**Comment by akubak** [comment]
@ludeeus I forgot to push the fix that you've suggested (I've just done it). It should work now. Unfortunately, I still don't have a way to test the script other than running a vm with debian (I'll get that sorted in a couple of days).

**Comment by ludeeus** [comment]
@Antoni-K Any progress?

**Comment by ludeeus** [comment]
@Antoni-K any update on this?

**Comment by ludeeus** [comment]
Closing down this now, it can be revisited at a later stage.

**Pull request closed without merging**